azurerm/resource_arm_private_link_endpoint_test.go

```diff
@@ -0,0 +1,246 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMPrivateEndpoint_basic(t *testing.T) {
+	resourceName := "azurerm_private_link_endpoint.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateEndpoint_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateEndpointExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "subnet_id"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateEndpoint_complete(t *testing.T) {
+	resourceName := "azurerm_private_link_endpoint.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateEndpoint_complete(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateEndpointExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "subnet_id"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.env", "test"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMPrivateEndpoint_update(t *testing.T) {
+	resourceName := "azurerm_private_link_endpoint.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPrivateEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMPrivateEndpoint_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPrivateEndpointExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "subnet_id"),
+				),
+			},
+			{
```

Comment:
Could we add import checks between each step?

Reply:
Fixed.